### DISCUSSION TITLE
PytorchLightening trainer?

### DISCUSSION THREAD

**Discussion by wjgancn** [deepinv/deepinv#247] [General]
Hello,

Thank you for creating such an amazing repository. I am just curious if there are any plans to make this repository compatible with the PyTorch Lightning Trainer (https://lightning.ai/docs/pytorch/stable/api_references.html#trainer). The Lightning Trainer functions similarly to the existing trainer in this repository but offers many additional options, particularly useful for distributed training. I think a straightforward way to achieve compatibility could be instancing the model of torch.nn.Module in the LightningModule class.

Thanks!

**Comment by matthieutrs** [comment]
Hey @wjgancn, thanks for the kind words 😊 

At the moment, we are not planning to provide further support for torch lightning to avoid as much as possible additional dependencies (even optional). However, if you are keen using torch lightning, I think the current modules implementation should be compatible. Are you currently witnessing anything preventing you to use torch lightning with deepinv?

**Comment by Andrewwango** [comment]
Would [Lightning Fabric](https://lightning.ai/docs/fabric/stable/#why-fabric) be a good compromise?
